mlflow/spark.py

```diff
@@ -167,14 +167,22 @@ def log_model(
                       The model signature can be :py:func:`inferred <mlflow.models.infer_signature>`
                       from datasets with valid model input (e.g. the training dataset with target
                       column omitted) and valid model output (e.g. model predictions generated on
-                      the training dataset), for example:
+                      the training dataset). If the signature describes model output then the
+                      corresponding columns will be included in the output schema from a
+                      `predict()` transformation. For example:
 
-                      .. code-block:: python
+      .. code-block:: python
+
+        from mlflow.models.signature import infer_signature
+        train = df.drop("target_label")
+        predictions = ... # compute model predictions
+        signature = infer_signature(train, predictions)
+
+        # show how signature defines output schema in `predict` transformation
+        mlflow.spark.log_model(model, "model", signature=signature, registered_model_name="mymodel")
+        # predict will now return data corresponding to the output schema from the signature
+        print(mlflow.pyfunc.load_model("models:/mymodel/1").predict(predictions.toPandas()))
```

Comment:
```suggestion
        # define the output signature for returned predictions from a `predict()` transformation
        mlflow.spark.log_model(model, "model", signature=signature, registered_model_name="mymodel")
```

No need to put a print statement in the code block.